src/Util/EntityImportUtil.php

```diff
@@ -20,4 +64,101 @@ public function transformFieldMappingSourceValueToSelect($options)
+
+        if ($this->containerUtil->isBackend()) {
+            $this->handleDebugMessageForBackend();
+        }
+    }
+
+    public function handleImportError(string $message, EntityImportConfigModel $importConfig): void
+    {
+        $this->initErrorHandling($message);
+
+        if ($this->containerUtil->isBackend()) {
+            $this->handleDebugMessageForBackend();
+            $this->reloadPageWithoutAction();
+        } else {
+            $this->handleDebugMessageForFrontend($importConfig);
+
+            exit();
+        }
+    }
+
+    /**
+     * set up protected values and log action.
+     */
+    protected function initErrorHandling(string $message): void
+    {
+        $this->setProperties($message);
+        $this->logError();
+    }
+
+    protected function setProperties(string $message): void
+    {
+        $this->exception = $message;
+        $this->debugConfig = $this->getDebugConfig();
+    }
+
+    protected function handleDebugMessageForFrontend(EntityImportConfigModel $importConfig): void
+    {
+        if ($importConfig->errorNotificationLock) {
+            return;
+        }
+
+        if (isset($this->debugConfig['email']) && $this->debugConfig['email']) {
+            Controller::loadLanguageFile('default');
+            $content = sprintf('An error occurred on domain "%s"', $importConfig->cronDomain).' : '.$this->exception;
+            $email = new Email();
+            $email->html = $content;
+            $email->text = strip_tags($content);
+            $email->subject = sprintf($GLOBALS['TL_LANG']['MSC']['entityImport']['exceptionEmailSubject'], $importConfig->title, $importConfig->id);
+            $email->sendTo($importConfig->errorNotificationEmail ?: $GLOBALS['TL_CONFIG']['adminEmail']);
+        }
+
+        $this->databaseUtil->update('tl_entity_import_config', ['errorNotificationLock' => '1'], 'tl_entity_import_config.id=?', [$importConfig->id]);
+    }
+
+    /**
+     * add the error message to display in backend.
+     */
+    protected function handleDebugMessageForBackend(): void
+    {
+        Message::addError($this->exception);
+    }
+
+    protected function getDebugConfig(): ?array
+    {
+        $config = $this->container->getParameter('huh_entity_import');
+
+        return $config['debug'];
+    }
+
+    /**
+     * if set in the config log the error.
+     */
+    protected function logError(): void
+    {
+        if (isset($this->debugConfig['contao_log']) && $this->debugConfig['contao_log']) {
+            $this->containerUtil->log($this->exception, 'executeImport', LogLevel::ERROR);
+        }
+    }
+
+    /**
+     * reload if import was initialized from backend.
+     */
+    protected function reloadPageWithoutAction(): void
+    {
+        $url = $this->urlUtil->getCurrentUrl();
+
+        if (false === strpos($url, 'key=import')) {
+            return;
+        }
+
+        $url = str_replace('key=import&', '', $url);
+        $this->urlUtil->redirect($url);
```

Comment:
Please use UrlUtil::removeQueryString() for url manipulation. You don't need to pass in the current url since the method gets it on its own.